iop/fs/bdmfs_vfat/src/fat_driver.c

```diff
@@ -995,8 +995,9 @@ int fat_mount(struct block_device* bd)
     // Filter for supported partition IDs:
     // - 0x0b = FAT32 with CHS addressing
     // - 0x0c = FAT32 with LBA addressing
-    if (bd->parId != 0x0b && bd->parId != 0x0c)
-        return -1;
+    // AKuHAK: dont filter partition, fat_getPartitionBootSector() will care about that, this is filtering too much
+    // if (bd->parId != 0x0b && bd->parId != 0x0c)
+    //     return -1;
```

Comment:
You have not mentioned which problem you are trying to solve in this PR, but could it be this was the only change needed?

If there is no MBR partition found, then the FAT32 file system can mount the entire drive (usb0p0). I see now that this filter would then fail, becouse the parId is always 0 without an MBR.

The filter can be changed to something like this:
```
    // Filter for supported partition IDs:
    // - 0x00 = Possibly a removable drive
    // - 0x0b = FAT32 with CHS addressing
    // - 0x0c = FAT32 with LBA addressing
    if (bd->parId != 0x00 && bd->parId != 0x0b && bd->parId != 0x0c)
        return -1;
```

But it can also be completely removed as you proposed.
Also take a look here at all possible partition types:
https://en.wikipedia.org/wiki/Partition_type

Reply:
FAT16 and Extended also is supported. `fat_getPartitionBootSector` will proceed with necessary checks if this partition is really from the FAT family, so we don't need to care about that.